scipy/integrate/_cubature.py

```diff
@@ -347,3 +718,406 @@ def _process_subregion(data):
     err_sub = rule.estimate_error(f, a_k_sub, b_k_sub, args)
 
     return a_k_sub, b_k_sub, est_sub, err_sub
+
+
+def _is_strictly_in_region(point, a, b):
+    xp = array_namespace(point, a, b)
+
+    if xp.all(point == a) or xp.all(point == b):
+        return False
+
+    return xp.all(a <= point) and xp.all(point <= b)
+
+
+def _split_at_points(a, b, points):
```

Comment:
I think this can be done simpler... Don't you just need to sort and check for infinite points?

I think this can be done actually much simpler... Would also be good to avoid the loops I played around with it, something like the following? (Needs to be adapted).

`xp.concatenate([xp.expand_dims(xp.asarray(a), axis = 0), xp.sort(points, axis = 0), xp.expand_dims(xp.asarray(b), axis = 0)], axis = 0)`